### ISSUE TITLE
ENH: eventsource.archiver more flexible if not already

### ISSUE THREAD

**Issue by ambarb** [bluesky/databroker#401]
For `databroker` version `'0.9.4` & `0.11.1` it seems that `table_given_times()` is not available as imported.  Am I missing something?  I see it here: 
https://github.com/NSLS-II/databroker/blob/171c98543d7a06deee2d8518e54d4709653880b3/databroker/eventsource/archiver.py#L233

This is something that I would use, especially in a context like:
```python
from databroker.eventsource.archiver import ArchiverEventSource
arc = ArchiverEventSource(ARCHIVER_URL, 'US/Eastern', pv_dict)
hdr = db[7777]
arc.table_given_times(since=db.start.time-5*60, until=db.stop.time)
```

though, what I am doing now is just trying to plot some data from the archiver for long period of time. This is unrelated to data analysis, but I am trying to understand what is going on with equipment/instrumentation.  I need to process/analyze more than what CSStudio provides and it seemed silly to export the data from CSStudio to then turn around an import it into jupyter notebook.

**Comment by ambarb** [comment]
I am able to get the information i need, but please let me know how to enable the `tables_given_times()` and its dependencies.

```python
arch_addrs = ARCHIVER_URL + '/retrieval/data/getData.json'
_from = 1458964799
_to = 1458964799+60
times=[_from, _to]
timezone=pytz.timezone('US/Eastern')

times_correct = []
for t in times:
    t = datetime.fromtimestamp(t)
    times_correct.append(timezone.localize(t).replace(microsecond=0).isoformat())
_from, _to = times_correct
    
pv=arc.pvs['a']
params = {'pv': pv, 'from': _from, 'to': _to}
req=requests.get(arch_addrs, params,stream=True)

req.raise_for_status()

raw, = req.json()

secs = [x['secs'] for x in raw['data']]
nanos = [x['nanos'] for x in raw['data']]
data = [x['val'] for x in raw['data']]

asecs = np.asarray(secs)
ananos = np.asarray(nanos)
times = asecs*1.0e+3 + ananos*1.0e-6

datetimes = pd.to_datetime(times, unit='ms')``

**Comment by mrakitin** [comment]
The `table_given_times()` function was introduced via https://github.com/NSLS-II/databroker/pull/359/files#diff-92a7328ca978fe3c83efe34d6e3518bbR233, which was released in v0.12.0, so you may want to use a newer databroker to use the function.

**Comment by ambarb** [comment]
Thanks for the reassurance that the function is in a release.  Do I need to clone the current nsls2 analysis environment as my own and then conda update?  will that take care of it, or do I need to know the specific tag?

**Comment by ambarb** [comment]
Also, if I instruct the function that I want to use the Eastern TIme zone, it goes an pulls the appropriate data, but it always returns that data with a GMT time stamp.  Is the intended functionality?

**Comment by mrakitin** [comment]
> Also, if I instruct the function that I want to use the Eastern TIme zone, it goes an pulls the appropriate data, but it always returns that data with a GMT time stamp. Is the intended functionality?

Not sure exactly, it's a question to @malitsky regarding the implementation.

**Comment by mrakitin** [comment]
> Also, if I instruct the function that I want to use the Eastern TIme zone, it goes an pulls the appropriate data, but it always returns that data with a GMT time stamp. Is the intended functionality?

You should have the latest conda environments, which contain the latest databroker version (and the archiver features). Are you running in Jupyter? Which beamline? We may need to update the kernels then (@danielballan and I can handle it after we know more details).

**Comment by ambarb** [comment]
This is at CSX using srv1 current.  I confirmed before replying.  Databroker is version `0.11.1`

**Comment by ambarb** [comment]
I think SIX is also like this

**Comment by mrakitin** [comment]
@ambarb, we with @danielballan plan to update the kernels at CSX first and perform the testing, namely, creating a legacy tag with the matching version of the conda env used, and add a new kernel which points to the latest analysis environments. If this succeeds, we will perform the same update for SIX kernels.

When is a good time to do that, so we don't disrupt users? Do you want to also participate in this activity, so we can test together and make sure we break nothing (especially old notebooks)? I can come to CSX or SIX tomorrow.

**Comment by ambarb** [comment]
@mrakitin @danielballan  I spoke with most everyone and I think we will hold off on updates until a maintenance day or the conda env for 2019-3 analysis.  Thanks for the help and the clarification.

I want to close this issue, but I will wait for @malitsky to comment about GMT timestamps on the return of the data if asked for in EST.  Then we can decide to open and ENH issue or not.

**Comment by malitsky** [comment]
@ambarb, it will be looked over/updated tomorrow/Monday.

**Comment by malitsky** [comment]
a flexible data retrieval interface for arbitrary time intervals is supported by Archiver Reader in the updated ARVPYF version (https://github.com/SciIO/arvpyf). @ambarb, ARVPYF requires only the Requests module and can be directly installed with pip. Explicit time zone conversion was included in the databroker PR (#405).

**Comment by ambarb** [comment]
That is great.  Easy install instruction and the documentation looks great.
